client/dom/use-object-url.tsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react'
+import { useValueAsRef } from '../state-hooks'
+
+/**
+ * A hook which creates an object URL out of the received file, while also making sure that stale
```

Comment:
This hook does not make sure that object URLs are revoked, it only revokes if the URL *changes*. So you need to fix that